docs/docusaurus/docs/conceptual_guides/metricproviders.md

```diff
@@ -0,0 +1,105 @@
+
+## Class hierarchy
+
+Although the class hierarchy for MetricProviders and Expectations is different, they use the same naming conventions. The following is the MetricProviders class hierarchy:
+
+```text
+MetricProvider
+    QueryMetricProvider
+    TableMetricProvider
+        ColumnAggregateExpectation
+    MapMetricProvider
+        ColumnMapMetricProvider
+            RegexColumnMapMetricProvider
+            SetColumnMapMetricProvide
+        ColumnPairMapMetricProvider
+        MulticolumnMapExpectation
+        MulticolumnMapMetricProvider
+```
+
+If you’re not sure which MetricProvider you should use with an Expectation, you can usually infer the correct MetricProvider class name from the Expectation class name. For example:
+
+| Expectation class                 | MetricProvider class                    |
+| --------------------------------- | --------------------------------------- |
+| Expectation                       | MetricProvider                          |
+| BatchExpectation                  | TableMetricProvider                     |
+| ColumnAggregateExpectation        | ColumnAggregateMetricProvider           |
+| ColumnMapExpectation              | ColumnMapMetricProvider                 |
+| RegexBasedColumnMapExpectation    | RegexColumnMapMetricProvider (built in) |
+| SetBasedColumnMapExpectation      | SetColumnMapMetricProvider (built in)   |
+| ColumnPairMapExpectation          | ColumnPairMapMetricProvider             |
+| MulticolumnMapExpectation         | MulticolumnMapMetricProvider            |
+| QueryExpectation                  | QueryMetricProvider                     |
+
+
+Sometimes, the MetricProvider class is created directly from the Expectation class, so you don’t need to specify a MetricProvider or methods when declaring a new Expectation. For example, the RegexColumnMapMetricProvider and the SetColumnMapMetricProvider classes.
+
+## Define Metrics with a MetricProvider
+
+The API for MetricProvider classes is unusual. MetricProvider classes are never intended to be instantiated, and they don’t have inputs or outputs in the normal sense of method arguments and return values. Instead, the inputs for MetricProvider classes are methods for calculating the Metric on different backend applications. Each method must be decorated with an appropriate decorator. On `new`, the class registers the methods as part of the Metrics registry so that they can be invoked as Metrics. The registered Metrics are the only output from MetricProviders.
+
+:::note
+Decorators invoked on `new` can make maintainability challenging. GX intends to address this shortcoming in future releases.
+:::
+
+A typical MetricProvider class has three methods that correspond to the three primary ExecutionEngines used by GX (pandas, SQL and SparkDF). Each of the three methods are marked with an appropriate decorator.
```

Comment:
```suggestion
A typical MetricProvider class has three methods that correspond to the three primary ExecutionEngines used by GX (pandas, SQL, and SparkDF). Each of the three methods are marked with an appropriate decorator.
```
It may be a good idea to include a bullet list here of (backend: decorator) pairs.  e.g:
```
- pandas: `@pandas_decorator_whatever_it_is`
```